src/Learning/KWData/KWClass.h

```diff
@@ -524,13 +532,32 @@ class KWClass : public Object
 	void InitializeRandomRuleParameters(KWDerivationRule* rule, const ALString& sAttributeName,
 					    int& nRuleRankInAttribute);
 
-	// Verification de l'integrite de la classe en ce qui concerne sa composition
-	// Il ne doit pas y avoir de cycle dans le graphe des utilisation entre classes par composition
-	// La taille des cles doit etre croissante avec la profondeur d'utilisation dans la composition
-	// L'attribut parent en parametre (potentiellement NULL) fournit des informations sur la profondeur
-	// dans l'arbre de composition, ce qui permet de tester la coherence de longueur des cle
-	// Le dictionnaire de classes en parametre permet de detecter les cycles
-	boolean CheckClassComposition(KWAttribute* parentAttribute, NumericKeyDictionary* nkdComponentClasses) const;
+	// Verification de l'integrite de la classe en ce qui concerne sa composition native,
+	// c'est a dire de sa hierarchie induite par l'ensemble des attributs relations non calcules,
+	// et si demande la coherence de son utilisation des cles dans la hierarchie pour permettre
+	// de rendre les donnees stockable sur une base multi-table a base de fichiers
+	//
+	// Il ne doit pas y avoir de cycle dans le graphe des utilisations entre classes par composition
+	// Le parametre de verification des cles, avec ou sans message d'erreur, permet de verifier qu'un
+	// dictionnaire est stockable sur disque via un mapping multi-fichier, au moyen de cles coherentes.
+	// - le dictionnaire doit avoir une cle s'il contient des sous-tables
+	// - la taille des cles doit etre croissante avec la profondeur d'utilisation dans la composition
+	// Note que l'on peut avoir des dictionnaires non stockage dans le cas de regles de derivation de creation
+	// de table, qui peuvent exploiter des dictionnaire quelconques (non Root), avec ou sans cle
+	boolean CheckNativeComposition(boolean bCheckKeys, boolean bVerboseCheckKeys) const;
+
+	// Methode interne utilisee par CheckNativeComposition, avec des parametres techniques supplementaire
+	// permettant son implementation de facon recursive
+	// - parentAttribute: correspond a l'attribut utilisant la classe (NULL pour l'appel initial),
+	//   ce qui fournit des informations sur la profondeur dans l'arbre de composition, et qui
+	//   permet de tester la coherence de longueur des cles
+	// - nkdComponentClasses: dictionnaire de classes traitees permettant de detecter les cycles
```

Comment:
I would mention the "accumulator" nature of `nkdComponentClasses` (see previous comment related to this).

Reply:
Le concept d'accumulator qui t'es familier ne me "parle" pas.
Les paramètres sont ici documentés de façon similaire a l'ensemble des méthodes utilisant le même type de pattern, avec une méthode "publique" s'appuyant sur une méthode "interne" ayant un ou plusieurs paramètres de travail.
Et modifier le commentaire ici appellerait une recherche systématique de cet type de pattern pour préserver l'homogénéisation des commentaires.

Reply:
By "accumulator" I understand here a "working parameter" (paramètre de travail) of an "internal" recursive function, which is a container structure that "accumulates" data from one recursive call to another.
But Ok, I got your point. We can keep it thusly IMHO.